engines/experiment/factory.go

```diff
@@ -0,0 +1,44 @@
+package experiment
+
+import (
+	"encoding/json"
+
+	"github.com/gojek/turing/engines/experiment/manager"
+	"github.com/gojek/turing/engines/experiment/plugin"
+	"github.com/gojek/turing/engines/experiment/runner"
+	v1 "github.com/gojek/turing/engines/experiment/v1"
+	"github.com/mitchellh/mapstructure"
+	"go.uber.org/zap"
+)
+
+// EngineFactory interface defines methods for accessing manager/runner
+// of a given experiment engine
+type EngineFactory interface {
```

Comment:
Though we are combining the manager and runner here, they won't be initialised together, right? For example, EngineFactory's `GetExperimentRunner` call must be made for the runner to be initialised and there wont be any initialisation of the corresponding experiment manager within the router. Is my understanding correct?

Reply:
Yup, that's right.